core/src/main/java/hudson/util/CompoundEnumeration.java

```diff
@@ -21,6 +22,11 @@ public CompoundEnumeration(Enumeration... e) {
 
     public CompoundEnumeration(Iterable<Enumeration<? extends T>> e) {
         this.base = e.iterator();
+        if (this.base.hasNext()) {
+            this.cur = this.base.next();
+        } else {
+            this.cur = Collections.emptyEnumeration();
+        }
```

Comment:
fixes an an existing bug (as cur would always be `null` and `hasMoreElements` should have thrown a `NullPointerExcetpion`-  quick search shows this was never used before.

Reply:
> fixes an an existing bug (as cur would always be `null` and `hasMoreElements` should have thrown a `NullPointerExcetpion`- quick search shows this was never used before.

Yes, as I stated in the PR description:

> To implement `ClassLoader#getResources(String)` I used Kohsuke's `CompoundEnumeration` class, which appears to have never been used or tested before. It contained a bug, which I fixed.